key.core/src/test/java/de/uka/ilkd/key/speclang/njml/JMLParserExceptionTest.java

```diff
@@ -0,0 +1,199 @@
+import static org.junit.jupiter.api.Assertions.*;
+
+/**
+ * This test case is used to ensure that errors in JML (and perhaps also in Java)
+ * are reported with a reasonable error message and the right position pointing
+ * into the file.
+ *
+ * To add a test case, locate the "exceptional" directory in the resources
+ * (below the directory for this package here) and add your single Java file
+ * that contains an error that should be presented to the user (like syntax
+ * error, unresolved names, ...)
+ *
+ * The first lines of the Java file may contain meta data on what to expect
+ * from the exception. Meta data are key-value pairs like
+ *
+ * <pre>
+ * // &lt;key&gt;: &lt;value&gt;
+ * </pre>
+ *
+ * The following keys are supported
+ * <table>
+ * <tr>
+ * <th>Key</th>
+ * <th>Description</th>
+ * </tr>
+ * <tr>
+ * <td>{@code noException}</td>
+ * <td>This particular file must
+ * <b>not</b> throw an exception. Default: false</td>
+ * </tr>
+ * <tr>
+ * <td>{@code exceptionClass}</td>
+ * <td>Either a fully qualified class name or
+ * a short classname (w/o package prefix) of the actual type of the
+ * exception. Optional.</td>
+ * </tr>
+ * <tr>
+ * <td>{@code msgContains}</td>
+ * <td>A string which occur somewhere in the
+ * exception message (obtained via {@link Exception#getMessage()}). Optional</td>
+ * </tr>
+ * <tr>
+ * <td>{@code msgMatches}</td>
+ * <td>A regular expression that must match the exception message (obtained via
+ * {@link Exception#getMessage()}). Optional</td>
+ * </tr>
+ * <tr>
+ * <td>{@code msgIs}</td>
+ * <td>A string to which the exception message (obtained via {@link Exception#getMessage()}) must be
+ * equal. Optional</td>
+ * </tr>
+ * <tr>
+ * <td>{@code position}</td>
+ * <td>A tuple in form {@code <line>/<col>} describing the position within this file. Both line and
+ * column are 1-based.
+ * It is also checked that the URL of the location points to the file under test. Optional</td>
+ * </tr>
+ * <tr>
+ * <td>{@code ignore}</td>
+ * <td>Ignore this test case if set to true. Default is false.</td>
+ * </tr>
+ * <tr>
+ * <td>{@code broken}</td>
+ * <td>If broken tests are disabled, ignore this test case if set to true. Indicates that this needs
+ * to be fixed! Default is false.</td>
+ * </tr>
+ * <tr>
+ * <td>{@code verbose}</td>
+ * <td>Print the stacktrace if set to true. Default is false.</td>
+ * </tr>
+ * </table>
+ *
+ *
+ * @author Mattias Ulbrich
+ */
+public class JMLParserExceptionTest {
+
+    public static final boolean IGNORE_BROKEN = true;
+
+    private final static Pattern PROP_LINE =
+        Pattern.compile("//\\s*(\\p{Alnum}+)\\s*[:=]\\s*(.*?)\\s*");
+
+    public static Stream<Arguments> getFiles() throws URISyntaxException, IOException {
+        URL fileURL = JMLParserExceptionTest.class.getResource("exceptional");
+        assert fileURL != null : "Directory 'exceptional' not found";
+        assert fileURL.getProtocol().equals("file") : "Test resources must be in file system";
```

Comment:
Either use `getResource` but then do not expect a file system, or just use the file system directly. Test cases are executed in the corresponding project folder.

Reply:
I'd rather not rely on a particular layout of the data (i.e. the maven-layout with a resoureces folder). If there is such a directory in the classpath: great. If it is in a JAR file. ... does not work currently (and I do not want to implement it).
I'd argue that this is at least as flexible as your proposal, and error reporting is clear.